tensorrt_llm/executor/result.py

```diff
@@ -532,3 +565,58 @@ async def __anext__(self):
         except asyncio.TimeoutError:
             self._done = True
             raise StopAsyncIteration
+
+
+def compute_logprobs(
+    context_logits: Optional[torch.Tensor],
+    generation_logits: Optional[torch.Tensor],
+    output_token_ids: Optional[list[int]],
+    k_prompt_logprobs: int = 5,
+    k_logprobs: int = 5,
+) -> Tuple[Optional[LogprobResults], Optional[LogprobResults]]:
+    """
+    Compute top-K logprobs and ranks for each token position in prompt and generated texts.
+
+    Returns:
+        Tuple of:
+            - prompt_logprobs: list of dicts {token_id -> Logprob}
+            - generation_logprobs: list of dicts {token_id -> Logprob}
+    """
+
+    def _topk_logprobs(logits: torch.Tensor, top_k: int,
+                       tokens: Optional[list[int]]) -> LogprobResults:
+        if logits.dim() == 3 and logits.size(0) == 1:
+            logits = logits.squeeze(0)  # shape: [T, vocab]
+
+        logprobs = F.log_softmax(logits.to("cuda", dtype=torch.float32), dim=-1)
```

Comment:
Please add a `NOTE` here to mark the potential performance drop, as we may reuse this part and enhance in PyT sampler.

Reply:
Do you mean perf drop due to a few more additional operation?

Reply:
Do you mean perf drop due to a few more additional operation?

Reply:
I mean, the copy the logits tensor from host to cuda, we can record the reason here.